src/transformers/models/oneformer/processing_oneformer.py

```diff
@@ -76,13 +100,46 @@ def _preprocess_text(self, text_list=None, max_length=77):
         token_inputs = torch.cat(token_inputs, dim=0)
         return token_inputs
 
-    def __call__(self, images=None, task_inputs=None, segmentation_maps=None, **kwargs):
+    @staticmethod
+    def _validate_input_types(
+        images: Optional[ImageInput] = None,
+        task_inputs: Optional[PreTokenizedInput] = None,
+    ) -> None:
+        if task_inputs is None:
+            raise ValueError("You have to specify the task_inputs. Found None.")
+        elif images is None:
+            raise ValueError("You have to specify the images. Found None.")
+
+        if not isinstance(task_inputs, List) or not task_inputs:
+            raise TypeError("task_inputs should be a string or a list of strings.")
+
+        if not all(task in ["semantic", "instance", "panoptic"] for task in task_inputs):
+            raise ValueError("task_inputs must be semantic, instance, or panoptic.")
+
+    @staticmethod
+    def _add_args_for_backward_compatibility(args):
+        """
+        Remove this function once support for args is dropped in __call__
+        """
+        if len(args) > 2:
+            raise ValueError("Too many positional arguments")
+        return dict(zip(("task_inputs", "segmentation_maps"), args))
```

Comment:
not needed https://github.com/huggingface/transformers/pull/34578#discussion_r1859636312